mlflow/pyfunc/__init__.py

```diff
@@ -634,7 +634,7 @@ def __init__(
         if not model_meta:
             raise MlflowException("Model is missing metadata.")
         self._model_meta = model_meta
-        self._model_impl = model_impl
+        self.__model_impl = model_impl
```

Comment:
Why do we need to add `_`?

Reply:
ah I see

Reply:
So we can change that variable, but always ensure we're exposing the private access method (this is for RAG Studio :) )

Reply:
Good idea! Added :)

Reply:
Sorry, we can remove the comment. What I meant was like this:

```python
# `_model_impl` is defined as is defined as @developer_stable
self._model_impl = model_impl
```

I think a property is better than a comment.